orion/pkg/repos/class_repo_test.go

```diff
@@ -0,0 +1,271 @@
+		now,
+		now,
+		sql.NullTime{},
+		"program1",
+		"2020_spring",
+		"final_review",
+		"program1_2020_spring_final_review",
+		"churchill",
+		"3 pm - 5 pm",
+		now,
+		now.Add(time.Hour*24*60),
+	)
+	mock.ExpectPrepare("^SELECT (.+) FROM classes WHERE class_id=?").
+		ExpectQuery().
+		WithArgs("program1_2020_spring_final_review").
+		WillReturnRows(rows)
+	got, err := repo.SelectByClassId("program1_2020_spring_final_review") // Correct classId
+	if err != nil {
+		t.Errorf("Unexpected error %v", err)
+	}
+
+	// Validate results
+	want := domains.Class{
+		Id:         1,
+		CreatedAt:  now,
+		UpdatedAt:  now,
+		DeletedAt:  sql.NullTime{},
+		ProgramId:  "program1",
+		SemesterId: "2020_spring",
+		ClassKey:   "final_review",
+		ClassId:    "program1_2020_spring_final_review",
+		LocationId: "churchill",
+		Times:      "3 pm - 5 pm",
+		StartDate:  now,
+		EndDate:    now.Add(time.Hour * 24 * 60),
+	}
+	if !reflect.DeepEqual(got, want) {
+		t.Errorf("Values not equal: got = %v, want = %v", got, want)
+	}
+	if err := mock.ExpectationsWereMet(); err != nil {
+		t.Errorf("Unfulfilled expectations: %s", err)
+	}
+}
+
+//
+// Create
+//
+func TestInsertClass(t *testing.T) {
+	db, mock, repo := initClassTest(t)
+	defer db.Close()
+
+	// Mock DB statements and execute
+	now := time.Now().UTC()
+	result := sqlmock.NewResult(1, 1)
+	mock.ExpectPrepare("^INSERT INTO classes").
+		ExpectExec().
+		WithArgs(
+			sqlmock.AnyArg(),
+			sqlmock.AnyArg(),
+			"program1",
+			"2020_spring",
+			"final_review",
+			"program1_2020_spring_final_review",
+			"churchill",
+			"3 pm - 5 pm",
+			now,
+			now.Add(time.Hour*24*60),
+		).WillReturnResult(result)
+	class := domains.Class{
+		ProgramId:  "program1",
+		SemesterId: "2020_spring",
+		ClassKey:   "final_review",
+		ClassId:    "program1_2020_spring_final_review",
+		LocationId: "churchill",
+		Times:      "3 pm - 5 pm",
+		StartDate:  now,
+		EndDate:    now.Add(time.Hour * 24 * 60),
+	}
+	err := repo.Insert(class)
+	if err != nil {
+		t.Errorf("Unexpected error %v", err)
+	}
+
+	// Validate results
+	if err := mock.ExpectationsWereMet(); err != nil {
+		t.Errorf("Unfulfilled expectations: %s", err)
+	}
+}
+
+//
+// Update
+//
+func TestUpdateClass(t *testing.T) {
+	db, mock, repo := initClassTest(t)
+	defer db.Close()
+
+	// Mock DB statements and execute
+	now := time.Now().UTC()
+	result := sqlmock.NewResult(1, 1)
+	mock.ExpectPrepare("^UPDATE classes SET (.*) WHERE class_id=?").
+		ExpectExec().
+		WithArgs(
+			sqlmock.AnyArg(),
+			"program2",
+			"2020_summer",
+			"",
+			"program2_2020_summer",
+			"churchill",
+			"5 pm - 7 pm",
+			now,
+			now.Add(time.Hour*24*30),
+			"program1_2020_spring_final_review",
+		).WillReturnResult(result)
+	class := domains.Class{
+		ProgramId:  "program2",
+		SemesterId: "2020_summer",
+		ClassKey:   "",
+		ClassId:    "program2_2020_summer",
+		LocationId: "churchill",
+		Times:      "5 pm - 7 pm",
+		StartDate:  now,
+		EndDate:    now.Add(time.Hour * 24 * 30),
```

Comment:
probably better to use a separate variable
later := now.Add(time.Hour * 24 * 30
then use it here and above (226)